localgov_eu_cookie_compliance.module

```diff
@@ -217,3 +248,146 @@ function _localgov_eu_cookie_compliance_defer_tracker_scripts(array &$attachment
+  $formatter = $variables["element"]["#formatter"];
+  /** @var \Drupal\Core\Field\FieldItemList $field_item */
+  $field_item = $variables['element']['#items'];
+  /** @var \Drupal\media\Entity\Media $media */
+  $media = $variables['element']['#object'];
+  $media_id = $media->id();
+  $variables['attributes']['class'][] = "media-oembed-$media_id";
+  $field_value = $field_item->getValue();
+  if (isset($field_value) && !empty($field_value)) {
+    $field_value = reset($field_value);
+    $video_src = $field_value['value'];
+  }
+
+  // Get the oembed settings lines and split them out.
+  $oembed_settings = Drupal::config('localgov_eu_cookie_compliance.settings')->get('disabled_oembeds') ?? "";
+  $disabled_oembeds = explode("\n", trim($oembed_settings));
+
+  // Convert the domains with optional category into an associative
+  // array of domain to category.
+  $blocked_domains = [];
+  foreach ($disabled_oembeds as $embed) {
+    // Extract the category condition, if present.
+    $parts = explode(':', $embed);
+    if (count($parts) == 1) {
+      $condition = "";
+      $embed = $parts[0];
+    }
+    else {
+      $condition = $parts[0];
+      $embed = end($parts);
+    }
+    $blocked_domains[$embed] = $condition;
+  }
+
+  foreach ($variables["items"] as &$item) {
+    switch ($formatter) {
+      case 'oembed':
+        // Get the video src being embedded.
+        $item["content"]["#attributes"]['data-src'] = $video_src;
+        $parsed = parse_url($video_src);
+
+        // Check against each rule.
+        foreach ($blocked_domains as $domain => $condition) {
+          // Convert the domain specification to a regex by:
+          // 1. Trim and quote the entered domain.
+          // 2. Replace * and ? with .* and . respectively.
+          // 3. Wrap in a case-insensitive search.
+          $regex = str_replace(
+            "\\?",
+            ".",
+            str_replace(
+              "\\*",
+              ".*",
+              preg_quote(trim($domain))
+            )
+          );
+          $regex = "/" . $regex . "/i";
+
+          // If the built regex matches, replace this item.
+          if (preg_match($regex, $parsed["host"])) {
+            // Add a class to the generated video markup.
+            $video = localgov_eu_cookie_compliance_preprocess_field_item_oembed($item);
+            // Render a link instead.
+            localgov_eu_cookie_compliance_preprocess_field_item_link($item);
+            // Record the video markup in the page JSON along with the condition.
+            $elements = [
+              $media->id() => [
+                'video' => Drupal::service('renderer')->render($video, FALSE),
+                'category' => $condition,
+              ],
+            ];
+            // Do not process more rules if one matched.
+            break;
+          }
+        }
+        // End of oembed case.
+        break;
+
+      // Not an oembed.
```

Comment:
Since we are dealing with only one type of formatter (i.e. "oembed") here, do we even need a `switch` statement?